### DISCUSSION TITLE
gripper simulation enviornment modeling

### DISCUSSION THREAD

**Discussion by 2-woo-10** [google-deepmind/mujoco#1945] [Asking for Help]
Hi, I'm a student and I'm trying to use MuJoCo for robot grasping

I'm looking for some help with simulation environment modeling .

I am working on a project where I need to use the Franka Panda gripper in the MuJoCo simulator to grasp an object. Specifically, I want to use the gripper alone to hold an object and then apply external forces to the object while measuring the forces involved.

My question is: During the simulation, is it possible to have the object fixed in place while the gripper grasps it, and then switch the object to a freejoint to apply forces to it? I want to ensure that I can change the object's state after it has been grasped so that I can apply forces and measure the response accurately.

Thank you for your assistance!

**Comment by 0tae-87** [comment]
https://mujoco.readthedocs.io/en/latest/XMLreference.html#option-flag-equality

Hi! I know this is an old question and you might have already solved it, but I’m leaving this here in case it helps you or others facing the same issue.

In MuJoCo, it is not possible to dynamically change the joint type (e.g., from fixed to free) during runtime. However, you can achieve the exact same effect by using Equality Constraints, specifically the Weld Constraint.
